### PR TITLE
Fix inconsistent help dialogue in examples

### PR DESCRIPTION
**Pull request by stotko**
- Fix crash in some examples if calling without parameters, i.e. `./<example>`
- Fix unrecognized `-h` and/or `--help` arguments
- Fix inconsistent formatting of help dialogue

---
This change is [Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3472)

### CODE REVIEW

File: examples/cpp/DepthCapture.cpp [hunk 1]
```diff
@@ -84,12 +84,23 @@ class VisualizerWithDepthCapture
...
+
 int main(int argc, char *argv[]) {
     utility::SetVerbosityLevel(utility::VerbosityLevel::Debug);
-    if (argc < 2) {
-        PrintOpen3DVersion();
-        utility::LogInfo("Usage:");
-        utility::LogInfo("    > DepthCapture  [filename]");
+    if (argc == 1 ||
+        utility::ProgramOptionExistsAny(argc, argv, {"-h", "--help"}) ||
+        argc != 2) {
```
Comment (germanros1987) [inline examples/cpp/DepthCapture.cpp:102]: argc != 2 || argc == 1 ? Check it out

File: examples/cpp/FileDialog.cpp [hunk 2]
```diff
@@ -30,16 +30,24 @@
...
+    // clang-format on
+    utility::LogInfo("");
 }
 
 int main(int argc, char *argv[]) {
     using namespace open3d;
-    if (argc == 1) {
+    if (argc == 1 ||
+        utility::ProgramOptionExistsAny(argc, argv, {"-h", "--help"}) ||
+        argc != 2) {
```
Comment (germanros1987) [inline examples/cpp/FileDialog.cpp:46]: argc == 1 || argc != 2 ?

File: examples/cpp/Flann.cpp [hunk 3]
```diff
@@ -30,14 +30,24 @@
...
+    // clang-format on
+    utility::LogInfo("");
+}
+
+int main(int argc, char *argv[]) {
+    using namespace open3d;
+    using namespace flann;
+    if (argc == 1 ||
+        utility::ProgramOptionExistsAny(argc, argv, {"-h", "--help"}) ||
+        argc != 2) {
```
Comment (germanros1987) [inline examples/cpp/Flann.cpp:49]: same check.

File: examples/cpp/ISSKeypoints.cpp [hunk 4]
```diff
@@ -35,14 +35,26 @@
...
 
     utility::SetVerbosityLevel(utility::VerbosityLevel::Debug);
-    if (argc < 3) {
-        utility::LogInfo("Open3D {}", OPEN3D_VERSION);
-        utility::LogInfo("Usage:");
-        utility::LogInfo("\t> {} [mesh|pointcloud] [filename] ...\n", argv[0]);
+
+    if (argc == 1 ||
+        utility::ProgramOptionExistsAny(argc, argv, {"-h", "--help"}) ||
+        argc != 3) {
```
Comment (germanros1987) [inline examples/cpp/ISSKeypoints.cpp:56]: same check.

File: examples/cpp/DepthCapture.cpp (hunk 1 above)
Comment (stotko) [reply examples/cpp/DepthCapture.cpp]: Works as expected, but is a bit over-complicated. Simplified.

File: examples/cpp/FileDialog.cpp (hunk 2 above)
Comment (stotko) [reply examples/cpp/FileDialog.cpp]: Simplified.

File: examples/cpp/Flann.cpp (hunk 3 above)
Comment (stotko) [reply examples/cpp/Flann.cpp]: Simplified.

File: examples/cpp/ISSKeypoints.cpp (hunk 4 above)
Comment (stotko) [reply examples/cpp/ISSKeypoints.cpp]: Simplified.

**Review by germanros1987 — APPROVED** [review]

**Pull request merged**